homeassistant/components/plant/__init__.py

```diff
@@ -312,14 +312,25 @@ def _load_history_from_db(self):
         _LOGGER.debug("Initializing values for %s from the database", self._name)
         with session_scope(hass=self.hass) as session:
             query = (
-                session.query(States)
+                session.query(States, StateAttributes)
                 .filter(
                     (States.entity_id == entity_id.lower())
                     and (States.last_updated > start_date)
                 )
+                .outerjoin(
+                    StateAttributes,
+                    States.attributes_id == StateAttributes.attributes_id,
+                )
                 .order_by(States.last_updated.asc())
             )
-            states = execute(query, to_native=True, validate_entity_ids=False)
+            states = []
+            if results := execute(query, to_native=False, validate_entity_ids=False):
+                # After 2023.8 make state.to_native require StateAttributes
```

Comment:
Does it mean after 2023.8 we no longer try to fetch attributes from the State table?
If so, please make this clear in the PR description and add a comment to other code which can be simplified with the 2023.8 release.

Reply:
`plant` and `statistics` should use the history api.  It looks like you already agree with that per https://github.com/home-assistant/core/pull/68315#issuecomment-1072116361 so I'm going to remove I'm going to remove these comment since these will likely go away long before 2023.8